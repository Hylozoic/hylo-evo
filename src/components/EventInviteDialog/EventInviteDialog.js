import React, { Component } from 'react'
import cx from 'classnames'
import { bgImageStyle } from 'util/index'
import ModalDialog from 'components/ModalDialog'
import CheckBox from 'components/CheckBox'
import Button from 'components/Button'
import styles from './EventInviteDialog.scss'
import { humanResponse } from 'store/models/EventInvitation'
import TextInput from 'components/TextInput'

export default class EventInviteDialog extends React.PureComponent {
  state = {
    invitedIds: [],
    searchTerm: ''
  }

  toggleInvite = id => {
    const { invitedIds } = this.state

    if (invitedIds.includes(id)) {
      this.setState({
        invitedIds: invitedIds.filter(invitedId => invitedId !== id)
      })
    } else {
      this.setState({
        invitedIds: invitedIds.concat([id])
      })
    }
  }

  onSearchChange = ({ target: { value } }) => {
    this.setState({searchTerm: value})
    this.props.fetchPeople(value)
  }

  getFilteredInviteSuggestions = () => {
    const { eventInvitations } = this.props
    const { searchTerm } = this.state
    return this.props.people.filter(person => {
      return !eventInvitations.map(ei => ei.id).includes(person.id) &&
      person.name.toLowerCase().includes(searchTerm.toLowerCase())
    })
  }

  submit = () => {
    const { eventId, onClose } = this.props
    const { invitedIds } = this.state
    this.props.invitePeopleToEvent(eventId, invitedIds)
    onClose()
  }

  render () {
    const { invitedIds } = this.state
    const { onClose, eventInvitations } = this.props

    const filteredInviteSuggestions = this.getFilteredInviteSuggestions()

    const inviteButtonLabel = invitedIds.length === 0
      ? 'Select people to invite'
      : invitedIds.length === 1
        ? 'Invite 1 person'
        : `Invite ${invitedIds.length} people`

    return <ModalDialog key='event-invite-dialog'
      closeModal={onClose}
      modalTitle={`Invite`}
      showCancelButton={false}
      showSubmitButton={false}
      style={{width: '100%', maxWidth: '620px'}}>
      <div styleName='container'>
        <Search onChange={this.onSearchChange} />
        <div styleName='inviteSuggestions'>
          {filteredInviteSuggestions.map(invitee => <InviteeRow
            key={invitee.id}
            person={invitee}
            selected={invitedIds.includes(invitee.id)}
            onClick={() => this.toggleInvite(invitee.id)}
          />)}
        </div>
        <div styleName='alreadyInvitedLabel'>Already Invited</div>
        <div styleName='alreadyInvited'>
          {eventInvitations.map(eventInvitation => <InviteeRow person={eventInvitation} showResponse key={eventInvitation.id} />)}
        </div>
        <Button
          small
          styleName='inviteButton'
          label={inviteButtonLabel}
          onClick={this.submit}
          disabled={invitedIds.length === 0} />
      </div>
    </ModalDialog>
  }
}

export function InviteeRow ({person, selected, showResponse, onClick}) {
  const { name, avatarUrl, response } = person

  return <div styleName={cx('row')} onClick={onClick}>
    <div styleName='col'>
      <div styleName='avatar' style={bgImageStyle(avatarUrl)} />
    </div>
    <div styleName='col'>
      {name}
    </div>
    {!showResponse && <div styleName='col check'>
      <CheckBox checked={selected} noInput />
    </div>}
    {showResponse && response && <div styleName='col response'>
      {humanResponse(response)}
    </div>}
  </div>
}

export class Search extends Component {
  render () {
    const { onChange } = this.props

    return <div styleName='search'>
      <TextInput theme={styles}
        inputRef={x => x && x.focus()}
        placeholder='Search members'
        onChange={onChange} />
    </div>
  }
}
