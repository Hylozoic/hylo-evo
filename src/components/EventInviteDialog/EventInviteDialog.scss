$avatar-size: 20px;

.container {
  height: 450px;
  display: flex;
  flex-direction: column;
}

.inviteSuggestions {
  max-height: 300px;
  overflow-y: auto;
}

.alreadyInvitedLabel {
  composes: bdy-lt-lg from 'css/typography.scss';
  margin-top: 10px;
}

.alreadyInvited {
  overflow-y: auto; 
}

.row {
  display: flex;
  align-items: left;
  border-bottom: solid 1px $color-cape-cod-10;
  width: 100%;
  cursor: pointer;
  &.selected {
    color: $color-rhino;
    background-color: $color-seashell;
  }
}

.col {
  display: flex;
  align-items: center;
  justify-content: left;
  padding: 5px;
}

.response {
  margin-left: auto;
  font-size: 12px;
}

.check {
  margin-left: auto;
}

.avatar {
  @include image($avatar-size);
  border-radius: $avatar-size/2;
}

.inviteButton {
  margin-top: auto;
  margin-left: auto;  
}

// Search
.search {
  margin-bottom: 10px;
}

// TextInput Theme
.wrapper {
  flex: 1;
  border: none;
  padding: none;
}

.input {
  color: $color-rhino;
  font-size: 34px;
  &::placeholder {
    color: $color-rhino-30;
  }
}

.clear {
  font-size: 18px;
}
