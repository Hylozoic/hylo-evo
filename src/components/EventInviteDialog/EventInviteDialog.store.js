import { createSelector } from 'redux-orm'
import orm from 'store/models'
import { pick } from 'lodash/fp'

export const MODULE_NAME = 'EventInviteDialog'
export const INVITE_PEOPLE_TO_EVENT = `${MODULE_NAME}/INVITE_PEOPLE_TO_EVENT`
export const INVITE_PEOPLE_TO_EVENT_PENDING = `${INVITE_PEOPLE_TO_EVENT}_PENDING`

export function presentPerson (person) {
  return {
    ...pick([ 'id', 'name', 'avatarUrl' ], person.ref),
    community: person.memberships.first()
      ? person.memberships.first().community.name : null
  }
}

export const peopleSelector = createSelector(
  orm,
  state => state.orm,
  session => session.Person.all()
    .orderBy('name')
    .toModelArray()
    .map(presentPerson)
)

export function invitePeopleToEvent (eventId, inviteeIds) {
  return {
    type: INVITE_PEOPLE_TO_EVENT,
    graphql: {
      query: `mutation ($eventId: ID, $inviteeIds: [ID]) {
        invitePeopleToEvent(eventId: $eventId, inviteeIds: $inviteeIds) {
          id
          eventInvitations {
            total
            hasMore
            items {
              id
              response
              person {
                id
                name
                avatarUrl
              }      
            }       
          } 
        }
      }`,
      variables: { eventId, inviteeIds }
    },
    meta: {
      eventId,
      inviteeIds,
      optimistic: true,
      extractModel: 'Post'
    }
  }
}
