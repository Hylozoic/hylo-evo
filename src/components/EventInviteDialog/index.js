import connector from './EventInviteDialog.connector'
import component from './EventInviteDialog'
export default connector(component)
