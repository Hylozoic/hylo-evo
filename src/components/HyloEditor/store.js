import { createSelector } from 'reselect'
import { fromJS } from 'immutable'
import { defaultSuggestionsFilter } from 'draft-js-mention-plugin'
import sampleMentions from './sampleMentions'

const FIND_MENTIONS = 'hyloEditor/FIND_MENTIONS'
const CLEAR_MENTIONS = 'hyloEditor/CLEAR_MENTIONS'

// Action Creators

export function findMentions (searchText) {
  return {
    type: FIND_MENTIONS,
    payload: { searchText }
  }
}

export function clearMentions (searchText) {
  return { type: CLEAR_MENTIONS }
}

// Reducer

export default function reducer (state = {mentionResults: fromJS([])}, action) {
  const { error, type, payload } = action
  if (error) return state

  switch (type) {
    case FIND_MENTIONS:
      return {...state, mentionResults: defaultSuggestionsFilter(payload.searchText, fromJS(sampleMentions))}
    case CLEAR_MENTIONS:
      return {...state, mentionResults: fromJS([])}
    default:
      return state
  }
}

// Selectors

export const getMentionResults = createSelector(
  (state) => state.hyloEditor,
  (state, props) => state.mentionResults
)
