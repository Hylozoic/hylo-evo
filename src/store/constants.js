export const _PENDING = '_PENDING'

export const ADD_USER_TYPING = 'ADD_USER_TYPING'
export const ADD_MODERATOR = 'ADD_MODERATOR'
export const ADD_MODERATOR_PENDING = ADD_MODERATOR + _PENDING
export const CHECK_LOGIN = 'CHECK_LOGIN'
export const CLEAR_USER_TYPING = 'CLEAR_USER_TYPING'
export const CLEAR_MODERATOR_SUGGESTIONS = 'CLEAR_MODERATOR_SUGGESTIONS'
export const CREATE_COMMENT = 'CREATE_COMMENT'
export const CREATE_COMMENT_PENDING = CREATE_COMMENT + _PENDING
export const CREATE_MESSAGE = 'CREATE_MESSAGE'
export const CREATE_MESSAGE_PENDING = CREATE_MESSAGE + _PENDING
export const CREATE_COMMUNITY = 'CREATE_COMMUNITY'
export const CREATE_COMMUNITY_PENDING = 'CREATE_COMMUNITY' + _PENDING
export const DROP_QUERY_RESULTS = 'DROP_QUERY_RESULTS'
export const EXTRACT_MODEL = 'EXTRACT_MODEL'
export const FETCH_COMMENTS = 'FETCH_COMMENTS'
export const FETCH_COMMUNITY_TOPIC = 'FETCH_COMMUNITY_TOPIC'
export const FETCH_MESSAGES = 'FETCH_MESSAGES'
export const FETCH_MESSAGES_PENDING = FETCH_MESSAGES + _PENDING
export const FETCH_THREAD = 'FETCH_THREAD'
export const FETCH_THREADS = 'FETCH_THREADS'
export const FETCH_TOPIC = 'FETCH_TOPIC'
export const FETCH_FEED_ITEMS = 'FETCH_FEED_ITEMS'
export const FETCH_NOTIFICATIONS = 'FETCH_NOTIFICATIONS'
export const FETCH_POST = 'FETCH_POST'
export const FETCH_POSTS = 'FETCH_POSTS'
export const FETCH_USER_SETTINGS = 'FETCH_USER_SETTINGS'
export const FETCH_MODERATOR_SUGGESTIONS = 'FETCH_MODERATOR_SUGGESTIONS'
export const FETCH_COMMUNITY = 'FETCH_COMMUNITY'
export const LEAVE_COMMUNITY = 'LEAVE_COMMUNITY'
export const LOGIN = 'LOGIN'
export const LOGOUT = 'LOGOUT'
export const MARK_ACTIVITY_READ = 'MARK_ACTIVITY_READ'
export const MARK_ACTIVITY_READ_PENDING = MARK_ACTIVITY_READ + _PENDING
export const MARK_ALL_ACTIVITIES_READ = 'MARK_ALL_ACTIVITIES_READ'
export const MARK_ALL_ACTIVITIES_READ_PENDING = MARK_ALL_ACTIVITIES_READ + _PENDING
export const REMOVE_MODERATOR = 'REMOVE_MODERATOR'
export const REMOVE_MODERATOR_PENDING = REMOVE_MODERATOR + _PENDING
export const RESET_NEW_POST_COUNT = 'RESET_NEW_POST_COUNT'
export const RESET_NEW_POST_COUNT_PENDING = 'RESET_NEW_POST_COUNT' + _PENDING
export const SET_CONFIRM_BEFORE_CLOSE = 'SET_CONFIRM_BEFORE_CLOSE'
export const SET_STATE = 'SET_STATE'
export const SIGNUP = 'SIGNUP'
export const STORE_CLEAR_FEED_LIST = 'STORE_CLEAR_FEED_LIST'
export const TOGGLE_TOPIC_SUBSCRIBE = 'TOGGLE_TOPIC_SUBSCRIBE'
export const TOGGLE_TOPIC_SUBSCRIBE_PENDING = 'TOGGLE_TOPIC_SUBSCRIBE_PENDING'
export const UPDATE_THREAD_READ_TIME = 'UPDATE_THREAD_READ_TIME'
export const UPDATE_USER_SETTINGS = 'UPDATE_USER_SETTINGS'
export const UPDATE_USER_SETTINGS_PENDING = 'UPDATE_USER_SETTINGS_PENDING'
export const UPLOAD_IMAGE = 'UPLOAD_IMAGE'
export const UNLINK_ACCOUNT = 'UNLINK_ACCOUNT'
export const VOTE_ON_POST = 'VOTE_ON_POST'
export const VOTE_ON_POST_PENDING = VOTE_ON_POST + _PENDING
