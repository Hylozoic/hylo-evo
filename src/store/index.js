import { createStore } from 'redux'
import createMiddleware from './middleware'
import orm from './models'
import rootReducer, { combinedReducers } from './reducers'

export function getEmptyState () {
  return combinedReducers({
    orm: orm.getEmptyState()
  }, {type: ''})
}

export default function (history, req) {
  return createStore(rootReducer, getEmptyState(), createMiddleware(history, req))
}
