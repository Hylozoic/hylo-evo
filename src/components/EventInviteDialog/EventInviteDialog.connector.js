import { connect } from 'react-redux'
import fetchPeople from 'store/actions/fetchPeople'
import { invitePeopleToEvent, peopleSelector } from './EventInviteDialog.store'

export function mapStateToProps (state, props) {
  const people = peopleSelector(state, props)

  return {
    people
  }
}

export const mapDispatchToProps = {
  fetchPeople, invitePeopleToEvent
}

export default connect(mapStateToProps, mapDispatchToProps)
