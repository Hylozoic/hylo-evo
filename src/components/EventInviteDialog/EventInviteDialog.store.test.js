import { presentPerson, invitePeopleToEvent } from './EventInviteDialog.store'

describe('presentPerson', () => {
  it('extracts the right keys', () => {
    const peopleModel = {
      ref: {
        id: 1,
        name: 'jo',
        avatarUrl: 'jo.png',
        unusedField: 'yup'
      },
      memberships: {
        first: () => ({
          community: {
            name: 'coomunity'
          }
        })
      }
    }

    expect(presentPerson(peopleModel)).toMatchSnapshot()
  })
})

describe('invitePeopleToEvent', () => {
  it('matches snapshot', () => {
    expect(invitePeopleToEvent(1, [2, 3, 4])).toMatchSnapshot()
  })
})
