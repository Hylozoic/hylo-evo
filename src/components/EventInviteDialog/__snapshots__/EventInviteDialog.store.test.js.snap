// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invitePeopleToEvent matches snapshot 1`] = `
Object {
  "graphql": Object {
    "query": "mutation ($eventId: ID, $inviteeIds: [ID]) {
        invitePeopleToEvent(eventId: $eventId, inviteeIds: $inviteeIds) {
          id
          eventInvitations {
            total
            hasMore
            items {
              id
              response
              person {
                id
                name
                avatarUrl
              }      
            }       
          } 
        }
      }",
    "variables": Object {
      "eventId": 1,
      "inviteeIds": Array [
        2,
        3,
        4,
      ],
    },
  },
  "meta": Object {
    "eventId": 1,
    "extractModel": "Post",
    "inviteeIds": Array [
      2,
      3,
      4,
    ],
    "optimistic": true,
  },
  "type": "EventInviteDialog/INVITE_PEOPLE_TO_EVENT",
}
`;

exports[`presentPerson extracts the right keys 1`] = `
Object {
  "avatarUrl": "jo.png",
  "community": "coomunity",
  "id": 1,
  "name": "jo",
}
`;
